DataFormats/HcalDetId/src/HcalDetId.cc

```diff
@@ -26,9 +33,23 @@ HcalDetId::HcalDetId(const DetId& gen) {
 	 subdet!=HcalTriggerTower && subdet!=HcalOther)) {
       throw cms::Exception("Invalid DetId") << "Cannot initialize HcalDetId from " << std::hex << gen.rawId() << std::dec; 
     }  
+    uint32_t rawid = gen.rawId();
+    if ((rawid&kHcalIdFormat2)==0) {
+      int zsid = (rawid&kHcalZsideMask1)?(1):(-1);
```

Comment:
about 5 copy-pastes of essentially the same block of code in the same file is suboptimal.